### ISSUE TITLE
rule: nontable-no-border-spacing

### ISSUE THREAD

**Issue by purupurupu** [purupurupu/css-noop-checker#128]
## Rule ID
`nontable-no-border-spacing`

## What does this rule detect?
`border-spacing` only applies to elements with `display: table` or `display: inline-table`. Has no effect on other display types.

## Example: should warn
```html
<div style="border-spacing: 10px;">content</div>
```

## Example: should NOT warn
```html
<div style="display: table; border-spacing: 10px;">
  <div style="display: table-cell;">cell</div>
</div>
```

## Related CSS Properties
`display`, `border-spacing`

## References
- [CSS 2.2 §17.6.1](https://www.w3.org/TR/CSS22/tables.html#separated-borders): applies to table elements
- Could be considered an extension to existing `element-no-table-props`
